crates/bindings/src/table.rs

```diff
@@ -388,6 +388,30 @@ impl<Tbl: Table, Col: Index + Column<Table = Tbl>> UniqueColumn<Tbl, Col::ColTyp
         let buf = IterBuf::take();
         update::<Tbl>(Col::index_id(), new_row, buf)
     }
+
+    /// Attempts to insert `new_row` into the table. If it exists, deletes the row with the same value in the unique column.
+    #[track_caller]
+    #[cfg(feature = "unstable")]
+    pub fn try_upsert(&self, new_row: Tbl::Row) -> Result<Tbl::Row, TryInsertError<Tbl>> {
+        let index = Col::get_field(&new_row);
+        match self._find(index) {
+            None => {
+                let buf = IterBuf::take();
+                insert::<Tbl>(new_row, buf)
+            }
+            Some(_existing) => Ok(self.update(new_row)),
+        }
```

Comment:
It would be more efficient (and semantically equivalent) to do a delete, ignore the result, and then do an insert.

Reply:
I didn't test it, but I'm not sure they're equivalent. Maybe we need the try_update as well to avoid panicking.

What do we want to do in this situation:

```rust
#[table(name = my_table_v1, public)]
pub struct MyTableV1 {
    #[primary_key]
    pub id: u64,

    #[unique]
    pub name: String,
}

// Suppose we already have 2 records in this table:
// {"id": 1, "name": "abc"}
// {"id": 2, "name": "def"}

#[reducer]
fn change_my_table_v1(ctx: &ReducerContext) {
  // This will cause the name to conflict.
  let row = MyTableV1 { id: 1, name: "def".to_string() };

  // Here we discard the Result because we want to commit it
  let _ = ctx.db.my_table_v1().id().try_insert_or_update(row);

  // At this point, the row with id 1 is deleted. Is that expected?
}
```